### DISCUSSION TITLE
How to imlement minting with only private key not using InputTransactionData in @aptos-labs/wallet-adapter-react?

### DISCUSSION THREAD

**Discussion by CryptainGolD** [aptos-labs/aptos-developer-discussions#502] [Questions] [answered]
Hello all, hope you are doing well
I'm in development of telegram bot for minting nfts
So I don't need `@aptos-labs/wallet-adapter-react` in `entry-functions/mint_nfts.ts` and I can use private key for this.

```
import { InputTransactionData } from "@aptos-labs/wallet-adapter-react";
import { MODULE_ADDRESS } from "@/constants";

export type MintNftArguments = {
  collectionId: string;
  amount: number;
};

export const mintNFT = (args: MintNftArguments): InputTransactionData => {
  const { collectionId, amount } = args;
  return {
    data: {
      function: `${MODULE_ADDRESS}::LaunchB0xx::mint_nft`,
      typeArguments: [],
      functionArguments: [collectionId, amount],
    },
  };
};
```

**Comment by 0xbrw** [comment]
you can create an account from the private key with something like this: https://github.com/aptos-labs/aptos-ts-sdk/blob/53a066f5c863253b1de57b2a83e9fd549bce82fe/examples/typescript/swap.ts#L209

cc @gregnazario might have more context here

**Comment by 0xmaayan** [comment, marked as answer]
To mint an nft, a user needs a wallet that holds their private key. I am not sure how you'd let users mint an nft without connecting their wallet (obviously users will not provide their private key for you).

If you use your own account to mint the nft, and you know its private key, you can generate an account based on the known private key, and then instead of calling the wallet `signAndSubmitTransaction` you simply use the SDK directly to submit the mint transaction
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/unit/account.test.ts#L96-L134
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/tests/e2e/transaction/transactionSubmission.test.ts#L432-L446
```
const privateKey = new Ed25519PrivateKey(privateKeyBytes);
const account = Account.fromPrivateKey({ privateKey });

const aptosConfig = new AptosConfig({network: Network})
const aptos = new Aptos(aptosConfig)
const transaction = await aptos.transaction.build.simple({
  sender: account.accountAddress,
  data: {
      function: `${MODULE_ADDRESS}::LaunchB0xx::mint_nft`,
      functionArguments: [collectionId, amount],
   },
 });
const response = await aptos.signAndSubmitTransaction({
 signer: account,
 transaction,
});
await aptos.waitForTransaction({
     transactionHash: response.hash,
});
```

You can take a look at our `TG mini app template` where you can build a TG mini app and let users submit a transaction on your app.  This mini app uses the wallet adapter for the user to connect their wallet and submit transaction with https://learn.aptoslabs.com/en/dapp-templates/telegram-mini-app-template

**Reply by CryptainGolD** [reply]
Thanks for your help really. it works now
